search-java/src/main/java/uk/os/elements/search/android/providers/addresses/AddressesProvider.java

```diff
@@ -0,0 +1,171 @@
+            mQueryPostcode = value;
+            return this;
+        }
+
+        public Builder queryUprn(boolean value) {
+            mQueryUprn = value;
+            return this;
+        }
+
+        public Builder queryFind(boolean value) {
+            mQueryFind = value;
+            return this;
+        }
+
+        public Builder queryBoundingBox(boolean value) {
+            mQueryBoundingBox = value;
+            return this;
+        }
+
+        public Builder queryRadius(boolean value) {
+            mQueryRadius = value;
+            return this;
+        }
+
+        public Builder queryNearest(boolean value) {
+            mQueryNearest = value;
+            return this;
+        }
+
+        public AddressesProvider build() {
+            return new AddressesProvider(mKey, mAddressApi, mQueryBoundingBox, mQueryFind, mQueryNearest,
+                    mQueryPostcode, mQueryRadius, mQueryUprn);
+        }
+    }
+
+    private final AddressApi mAddressApi;
+    private final String mKey;
+    private final boolean mQueryPostcode;
+    private final boolean mQueryUprn;
+    private final boolean mQueryFind;
+    private final boolean mQueryBoundingBox;
+    private final boolean mQueryRadius;
+    private final boolean mQueryNearest;
+
+    private AddressesProvider(String sKeyOpenNames, AddressApi addressApi, boolean queryBoundingBox, boolean queryFind,
+                             boolean queryNearest, boolean queryPostcode, boolean queryRadius, boolean queryUprn) {
+        mKey = sKeyOpenNames;
+        mAddressApi = addressApi;
+
+        mQueryBoundingBox = queryBoundingBox;
+        mQueryFind = queryFind;
+        mQueryNearest = queryNearest;
+        mQueryPostcode = queryPostcode;
+        mQueryRadius = queryRadius;
+        mQueryUprn = queryUprn;
+    }
+
+    public Observable<List<SearchResult>> query(String searchTerm) {
+        List<Observable<ServerResponse>> responses = queryProviders(searchTerm);
+        return Transform.toSearchResults(responses).map(deduplicate()).defaultIfEmpty(new ArrayList<SearchResult>());
+    }
+
+    public Observable<List<SearchResult>> query(String searchTerm, double lat, double lon) {
+        List<Observable<ServerResponse>> responses = queryProviders(searchTerm);
+        responses.addAll(queryProviders(lat, lon));
+
+        return Transform.toSearchResults(responses).map(deduplicate()).defaultIfEmpty(new ArrayList<SearchResult>());
+    }
+
+    private List<Observable<ServerResponse>> queryProviders(String searchTerm) {
+        List<Observable<ServerResponse>> list = new ArrayList<>();
+
+        if (mQueryFind) {
+            list.add(mAddressApi.find(mKey, searchTerm));
+        }
+        if (mQueryPostcode && GeoPattern.isPostcodeCandidate(searchTerm)) {
+            list.add(mAddressApi.postcode(mKey, searchTerm));
+        }
+        if (mQueryUprn && GeoPattern.isUprnCandidate(searchTerm)) {
+            list.add(mAddressApi.uprn(mKey, searchTerm));
+        }
```

Comment:
I assumed that folk could choose any combination.  For example:

```java
e.g. new AddressesProvider
                    .Builder(apiKey, provideAddressSearchApi())
                    .queryUprn(true)
                    .queryPostcode(true)
                    .queryFind(false)
                    .build()
```

The above would include UPRN and postcode buckets _but_ not the find bucket.  

The `isXCandidate` checks avoid 400 responses from user input not conforming to endpoint requirements.  This prevents the stream from exploding when the postcode bucket is included _but_ we are searching for "London Road", which would result in no results / error.

Clearly, an omnibox would benefit from a single endpoint with appropriate filters / param config.